### ISSUE TITLE
Compile errors in Mac OS X 10.12 Sierra

### ISSUE THREAD

**Issue by apoloval** [openMSX/debugger#57]
The `make` command fails with the following output:

```
>  make
Generating ui_BitMapViewer.h...
Generating ui_BreakpointDialog.h...
Generating ui_ConnectDialog.h...
Generating ui_GotoDialog.h...
Generating ui_PreferencesDialog.h...
Generating ui_SymbolManager.h...
Generating ui_VDPCommandRegisters.h...
Generating ui_VDPRegistersExplained.h...
Generating ui_VDPStatusRegisters.h...
Up to date: derived/config/Version.ii
Compiling BitMapViewer.cpp...
In file included from src/BitMapViewer.cpp:1:
In file included from src/BitMapViewer.h:4:
In file included from derived/src/ui_BitMapViewer.h:12:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/QVariant:1:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qvariant.h:39:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qlist.h:1026:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qstringlist.h:40:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qregexp.h:41:
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:739:55: error: no type named 'u16string' in namespace 'std'
    static inline QString fromStdU16String(const std::u16string &s);
                                                 ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:740:17: error: no type named 'u16string' in namespace 'std'
    inline std::u16string toStdU16String() const;
           ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:741:55: error: no type named 'u32string' in namespace 'std'
    static inline QString fromStdU32String(const std::u32string &s);
                                                 ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:742:17: error: no type named 'u32string' in namespace 'std'
    inline std::u32string toStdU32String() const;
           ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1307:53: error: no type named 'u16string' in namespace 'std'
inline QString QString::fromStdU16String(const std::u16string &s)
                                               ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1310:13: error: no type named 'u16string' in namespace 'std'
inline std::u16string QString::toStdU16String() const
       ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1311:15: error: no member named 'u16string' in namespace 'std'
{ return std::u16string(reinterpret_cast<const char16_t*>(utf16()), length()); }
         ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1313:53: error: no type named 'u32string' in namespace 'std'
inline QString QString::fromStdU32String(const std::u32string &s)
                                               ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1316:13: error: no type named 'u32string' in namespace 'std'
inline std::u32string QString::toStdU32String() const
       ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1318:19: error: expected ';' after expression
    std::u32string u32str(length(), char32_t(0));
                  ^
                  ;
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1318:10: error: no member named 'u32string' in namespace 'std'
    std::u32string u32str(length(), char32_t(0));
    ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1318:20: error: use of undeclared identifier 'u32str'
    std::u32string u32str(length(), char32_t(0));
                   ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1319:75: error: use of undeclared identifier 'u32str'
    int len = toUcs4_helper(d->data(), length(), reinterpret_cast<uint*>(&u32str[0]));
                                                                          ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1320:5: error: use of undeclared identifier 'u32str'
    u32str.resize(len);
    ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qstring.h:1321:12: error: use of undeclared identifier 'u32str'
    return u32str;
           ^
In file included from src/BitMapViewer.cpp:1:
In file included from src/BitMapViewer.h:4:
In file included from derived/src/ui_BitMapViewer.h:12:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/QVariant:1:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qvariant.h:41:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qmap.h:40:
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qpair.h:62:44: error: no member named 'declval' in namespace 'std'
        Q_DECL_NOEXCEPT_EXPR(noexcept(std::declval<T1&>() = p.first) && noexcept(std::declval<T2&>() = p.second))
                                      ~~~~~^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qcompilerdetection.h:1027:43: note: expanded from macro 'Q_DECL_NOEXCEPT_EXPR'
# define Q_DECL_NOEXCEPT_EXPR(x) noexcept(x)
                                          ^
In file included from src/BitMapViewer.cpp:1:
In file included from src/BitMapViewer.h:4:
In file included from derived/src/ui_BitMapViewer.h:12:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/QVariant:1:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qvariant.h:41:
In file included from /usr/local/Cellar/qt55/5.5.1/include/QtCore/qmap.h:40:
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qpair.h:62:52: error: 'T1' does not refer to a value
        Q_DECL_NOEXCEPT_EXPR(noexcept(std::declval<T1&>() = p.first) && noexcept(std::declval<T2&>() = p.second))
                                                   ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qpair.h:42:17: note: declared here
template <class T1, class T2>
                ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qpair.h:62:55: error: expected expression
        Q_DECL_NOEXCEPT_EXPR(noexcept(std::declval<T1&>() = p.first) && noexcept(std::declval<T2&>() = p.second))
                                                      ^
/usr/local/Cellar/qt55/5.5.1/include/QtCore/qpair.h:62:57: error: expected expression
        Q_DECL_NOEXCEPT_EXPR(noexcept(std::declval<T1&>() = p.first) && noexcept(std::declval<T2&>() = p.second))
                                                        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [derived/obj/BitMapViewer.o] Error 1
```

Suspecting C++11 flags are not properly set.

**Comment by apoloval** [comment]
I see this is how the compiler is invoked before reporting the errors:
```
c++ \
		 -MP -MMD -MF derived/dep/BitMapViewer.d \
		-o derived/obj/BitMapViewer.o -g -fPIC -pipe -Wall -I/usr/local/Cellar/qt55/5.5.1/include/QtCore -I/usr/local/Cellar/qt55/5.5.1/include/QtWidgets -I/usr/local/Cellar/qt55/5.5.1/include/QtGui -I/usr/local/Cellar/qt55/5.5.1/include/QtNetwork -I/usr/local/Cellar/qt55/5.5.1/include/QtXml -I/usr/local/Cellar/qt55/5.5.1/include -I/Library/Frameworks/QtCore.framework/Headers -I/Library/Frameworks/QtWidgets.framework/Headers -I/Library/Frameworks/QtGui.framework/Headers -I/Library/Frameworks/QtNetwork.framework/Headers -I/Library/Frameworks/QtXml.framework/Headers -Isrc -Isrc/openmsx -Iderived/config -Iderived/src -std=c++11 -mmacosx-version-min=10.7 -c src/BitMapViewer.cpp
```

Quite confusing, it's already using `-std=c++11`.

**Comment by apoloval** [comment]
Ok, what we were not using is the `-stdlib=libc++` flag to use the library C++11 shipped with Clang. Pull request ongoing.

**Comment by apoloval** [comment]
Cool! Fixed in #58.
